src/components/BusinessHours.vue

```diff
@@ -120,8 +120,74 @@ export default {
   },
   methods: {
     hoursChange: function(val) {
+      val = this.setClosingTimeBasedOnOpening(val)
+
       this.$emit('updated-hours', val);
-    }
+    },
+    calculateOpenTime(closeTime) {
+      const minutesPerHour = 60;
+      const additionalHours = 9;
+      const minHours = 0; // 12:00 AM is represented as 0 hours
+      const [hours, minutes] = [
+        parseInt(closeTime.substring(0, 2), 10),
+        parseInt(closeTime.substring(2, 4), 10),
+      ];
+
+      // Calculate the total minutes of the closing time
+      let totalMinutes = hours * minutesPerHour + minutes - (additionalHours * minutesPerHour);
+
+      // If the calculated open time is earlier than 12:00 AM, set it to 12:00 AM
+      if (totalMinutes < minHours * minutesPerHour) {
+        totalMinutes = minHours * minutesPerHour;
+      }
+
+      // Calculate the new hours and minutes for the open time
+      let newHours = Math.floor(totalMinutes / 60);
+      let newMinutes = totalMinutes % 60;
+
+      // Format the new open time as a string with padded zeros
+      return String(newHours).padStart(2, '0') + String(newMinutes).padStart(2, '0');
+    },
+    calculateCloseTime(openTime) {
+      const minutesPerHour = 60;
```

Comment:
if  `minutesPerHour ` `additionalHours ` are used in both functions. Wouln't be better to have them declared in data section?